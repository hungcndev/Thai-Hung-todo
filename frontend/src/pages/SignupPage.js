import React, { useState } from "react";
import { faEnvelope, faLock, faUser } from "@fortawesome/free-solid-svg-icons";
import { useAuthContext } from "../contexts/authStore";
import { useNavigate } from "react-router";
import axios from "axios";
import InputBox from "../components/InputBox/InputBox";
import NavBar from "../components/NavBar/NavBar";

export default function SignupPage() {
  const [info, setInfo] = useState({});
  const [error, setError] = useState("");
  const { setToken } = useAuthContext();
  const navigate = useNavigate();
  const onChangeHandler = (e) => {
    const { name, value } = e.currentTarget;
    setInfo({ ...info, [name]: value });
  };

  const submitHandler = async (e) => {
    e.preventDefault();
    try {
      if (info.confirmPassword !== info.password)
        throw Error(`Passwords don't match`);
      const { data } = await axios.post("/user/signup", {
        email: info.email,
        password: info.password,
      });
      setToken(data.token);
      axios.defaults.headers.common.authorization = `Bearer ${data.token}`;
      navigate("/");
    } catch (err) {
      setError(err?.response?.data?.message || err.message);
    }
  };

  return (
    <div>
      <NavBar />
      <div className="flex justify-center mt-32">
        <form action="submit" className="loginBox w-[450px]">
          <h1 className="font-sans font-bold text-center mb-10 text-3xl">
            CREATE AN ACCOUNT
          </h1>
          <InputBox
            name="username"
            placeholder={"Username"}
            icon={faUser}
            onChangeHandler={onChangeHandler}
          />
          <InputBox
            name="email"
            placeholder={"Email"}
            icon={faEnvelope}
            onChangeHandler={onChangeHandler}
          />
          <InputBox
            type="password"
            name="password"
            placeholder={"Password"}
            icon={faLock}
            onChangeHandler={onChangeHandler}
          />
          <InputBox
            type="password"
            name="confirmPassword"
            placeholder={"Confirm password"}
            icon={faLock}
            onChangeHandler={onChangeHandler}
          />
          {error.length > 0 && <div className="text-red-400 mb-5">{error}</div>}
          <button
            onClick={submitHandler}
            className="w-full py-3 px-7 mt-2 rounded-md text-xl font-semibold text-lg  bg-red text-white"
          >
            Register
          </button>
        </form>
      </div>
    </div>
  );
}
