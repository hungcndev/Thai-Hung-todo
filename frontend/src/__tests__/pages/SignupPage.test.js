import { render, cleanup } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import SignupPage from "../../pages/SignupPage";

afterEach(cleanup);

describe("Signup Page", () => {
  it("should render and match the snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <SignupPage />
      </BrowserRouter>
    );
    expect(
      asFragment(
        <BrowserRouter>
          <SignupPage />
        </BrowserRouter>
      )
    ).toMatchSnapshot();
  });
});
