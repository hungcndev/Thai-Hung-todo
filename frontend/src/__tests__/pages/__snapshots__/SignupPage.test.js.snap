// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Page should render and match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex justify-center mt-32"
    >
      <form
        action="submit"
        class="loginBox w-[450px]"
      >
        <h1
          class="font-sans font-bold text-center mb-10 text-3xl"
        >
          CREATE AN ACCOUNT
        </h1>
        <div
          class="flex border items-center mb-5 rounded-md"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-user ml-7 text-gray-500"
            data-icon="user"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256c70.7 0 128-57.31 128-128s-57.3-128-128-128C153.3 0 96 57.31 96 128S153.3 256 224 256zM274.7 304H173.3C77.61 304 0 381.6 0 477.3c0 19.14 15.52 34.67 34.66 34.67h378.7C432.5 512 448 496.5 448 477.3C448 381.6 370.4 304 274.7 304z"
              fill="currentColor"
            />
          </svg>
          <input
            class="ml-7 py-3 focus:outline-none flex-1 pr-5 font-medium text-base"
            name="username"
            placeholder="Username"
            required=""
            type=" text"
          />
        </div>
        <div
          class="flex border items-center mb-5 rounded-md"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-envelope ml-7 text-gray-500"
            data-icon="envelope"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 64C490.5 64 512 85.49 512 112C512 127.1 504.9 141.3 492.8 150.4L275.2 313.6C263.8 322.1 248.2 322.1 236.8 313.6L19.2 150.4C7.113 141.3 0 127.1 0 112C0 85.49 21.49 64 48 64H464zM217.6 339.2C240.4 356.3 271.6 356.3 294.4 339.2L512 176V384C512 419.3 483.3 448 448 448H64C28.65 448 0 419.3 0 384V176L217.6 339.2z"
              fill="currentColor"
            />
          </svg>
          <input
            class="ml-7 py-3 focus:outline-none flex-1 pr-5 font-medium text-base"
            name="email"
            placeholder="Email"
            required=""
            type=" text"
          />
        </div>
        <div
          class="flex border items-center mb-5 rounded-md"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-lock ml-7 text-gray-500"
            data-icon="lock"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M80 192V144C80 64.47 144.5 0 224 0C303.5 0 368 64.47 368 144V192H384C419.3 192 448 220.7 448 256V448C448 483.3 419.3 512 384 512H64C28.65 512 0 483.3 0 448V256C0 220.7 28.65 192 64 192H80zM144 192H304V144C304 99.82 268.2 64 224 64C179.8 64 144 99.82 144 144V192z"
              fill="currentColor"
            />
          </svg>
          <input
            class="ml-7 py-3 focus:outline-none flex-1 pr-5 font-medium text-base"
            name="password"
            placeholder="Password"
            required=""
            type="password"
          />
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-eye mr-7 cursor-pointer text-gray-500"
            data-icon="eye"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M160 256C160 185.3 217.3 128 288 128C358.7 128 416 185.3 416 256C416 326.7 358.7 384 288 384C217.3 384 160 326.7 160 256zM288 336C332.2 336 368 300.2 368 256C368 211.8 332.2 176 288 176C287.3 176 286.7 176 285.1 176C287.3 181.1 288 186.5 288 192C288 227.3 259.3 256 224 256C218.5 256 213.1 255.3 208 253.1C208 254.7 208 255.3 208 255.1C208 300.2 243.8 336 288 336L288 336zM95.42 112.6C142.5 68.84 207.2 32 288 32C368.8 32 433.5 68.84 480.6 112.6C527.4 156 558.7 207.1 573.5 243.7C576.8 251.6 576.8 260.4 573.5 268.3C558.7 304 527.4 355.1 480.6 399.4C433.5 443.2 368.8 480 288 480C207.2 480 142.5 443.2 95.42 399.4C48.62 355.1 17.34 304 2.461 268.3C-.8205 260.4-.8205 251.6 2.461 243.7C17.34 207.1 48.62 156 95.42 112.6V112.6zM288 80C222.8 80 169.2 109.6 128.1 147.7C89.6 183.5 63.02 225.1 49.44 256C63.02 286 89.6 328.5 128.1 364.3C169.2 402.4 222.8 432 288 432C353.2 432 406.8 402.4 447.9 364.3C486.4 328.5 512.1 286 526.6 256C512.1 225.1 486.4 183.5 447.9 147.7C406.8 109.6 353.2 80 288 80V80z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="flex border items-center mb-5 rounded-md"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-lock ml-7 text-gray-500"
            data-icon="lock"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M80 192V144C80 64.47 144.5 0 224 0C303.5 0 368 64.47 368 144V192H384C419.3 192 448 220.7 448 256V448C448 483.3 419.3 512 384 512H64C28.65 512 0 483.3 0 448V256C0 220.7 28.65 192 64 192H80zM144 192H304V144C304 99.82 268.2 64 224 64C179.8 64 144 99.82 144 144V192z"
              fill="currentColor"
            />
          </svg>
          <input
            class="ml-7 py-3 focus:outline-none flex-1 pr-5 font-medium text-base"
            name="confirmPassword"
            placeholder="Confirm password"
            required=""
            type="password"
          />
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-eye mr-7 cursor-pointer text-gray-500"
            data-icon="eye"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M160 256C160 185.3 217.3 128 288 128C358.7 128 416 185.3 416 256C416 326.7 358.7 384 288 384C217.3 384 160 326.7 160 256zM288 336C332.2 336 368 300.2 368 256C368 211.8 332.2 176 288 176C287.3 176 286.7 176 285.1 176C287.3 181.1 288 186.5 288 192C288 227.3 259.3 256 224 256C218.5 256 213.1 255.3 208 253.1C208 254.7 208 255.3 208 255.1C208 300.2 243.8 336 288 336L288 336zM95.42 112.6C142.5 68.84 207.2 32 288 32C368.8 32 433.5 68.84 480.6 112.6C527.4 156 558.7 207.1 573.5 243.7C576.8 251.6 576.8 260.4 573.5 268.3C558.7 304 527.4 355.1 480.6 399.4C433.5 443.2 368.8 480 288 480C207.2 480 142.5 443.2 95.42 399.4C48.62 355.1 17.34 304 2.461 268.3C-.8205 260.4-.8205 251.6 2.461 243.7C17.34 207.1 48.62 156 95.42 112.6V112.6zM288 80C222.8 80 169.2 109.6 128.1 147.7C89.6 183.5 63.02 225.1 49.44 256C63.02 286 89.6 328.5 128.1 364.3C169.2 402.4 222.8 432 288 432C353.2 432 406.8 402.4 447.9 364.3C486.4 328.5 512.1 286 526.6 256C512.1 225.1 486.4 183.5 447.9 147.7C406.8 109.6 353.2 80 288 80V80z"
              fill="currentColor"
            />
          </svg>
        </div>
        <button
          class="w-full py-3 px-7 mt-2 rounded-md text-xl font-semibold text-lg bg-xred hover:bg-red-500 text-white"
        >
          Register
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
